examples/ui-demo/src/hooks/useRecurringTransactions.ts

```diff
@@ -32,11 +33,13 @@ export type TransactionType = {
   state: TransactionStages;
   buyAmountUsdc: number;
   externalLink?: string;
+  timeToBuy?: number; // timestamp when the txn will initiate
+  secUntilBuy?: number; // seconds until the txn will initiate
```

Comment:
noob question: why do we need the timestamp here in addition to the countdown? could the countdown also serve as the time to reveal the transaction component's contents?

Reply:
The `timeToBuy` is used to calculate the number of seconds remaining each time the interval ticks. At first I was just trying to only have the `timeToBuy` and NOT include the `secToBuy` here, but I got a lot of annoying edge cases where it would be off by 1 due to re-renders, i.e. sometimes it would start from 11 sec and other times from 9 sec. This solution resulted in the cleanest UX.

Reply:
i think i found another good solution! will push shortly.